libs/payments/cart/src/lib/tax.service.ts

```diff
@@ -79,17 +103,31 @@ export class TaxService {
       });
 
     if (accountCustomer && accountCustomer.stripeCustomerId) {
-      const subscriptions = await this.subscriptionManager.listForCustomer(
-        accountCustomer.stripeCustomerId
+      const [subscriptions, customer] = await Promise.all([
+        this.subscriptionManager.listForCustomer(
+          accountCustomer.stripeCustomerId
+        ),
+        this.customerManager.retrieve(accountCustomer.stripeCustomerId),
+      ]);
+
+      const currentCurrency = String(customer.currency).toUpperCase();
+
+      const isTaxChangeAllowed = !(
+        subscriptions.length && currentCurrency !== locationCurrency
       );
 
       return {
-        isTaxLocked: !!subscriptions.length,
+        isTaxChangeAllowed,
+        status: isTaxChangeAllowed
+          ? TaxChangeAllowedStatus.Allowed
+          : TaxChangeAllowedStatus.CurrencyChange,
+        currentCurrency,
       };
     }
 
     return {
-      isTaxLocked: false,
+      isTaxChangeAllowed: true,
+      status: TaxChangeAllowedStatus.Allowed,
     };
```

Comment:
Is there a reason to return both a boolean value as well as the status itself?

I'd propose this method be renamed to `getTaxChangeStatus` and just have it return the status. I'd expect an "is" method to return a boolean value alone.

Reply:
Ah good point, nope, don't think so. I'll remove the boolean.